discord/guild.py

```diff
@@ -895,6 +896,64 @@ def get_member(self, user_id: int, /) -> Member | None:
         """
         return self._members.get(user_id)
 
+    _FETCHABLE = TypeVar(
+        "_FETCHABLE",
+        bound=Union[
+            VoiceChannel,
+            TextChannel,
+            ForumChannel,
+            StageChannel,
+            CategoryChannel,
+            Thread,
+            Member,
+        ],
+    )
+
+    async def get_or_fetch(
+        self, object_type: Type[_FETCHABLE], object_id: int, /
+    ) -> _FETCHABLE | None:
+        """Shortcut method to get data from guild object if it's cached or fetch from api if it's not.
+
+        Parameters
+        ----------
+        object_type: :class:`Type[VoiceChannel, TextChannel, ForumChannel, StageChannel, CategoryChannel, Thread, Member]`
+            Type of object to fetch or get.
+
+        object_id: :class:`int`
+            ID of object to get.
+
+        Returns
+        -------
+
+        Optional[:class:`~Type[VoiceChannel, TextChannel, ForumChannel, StageChannel, CategoryChannel, Thread, Member]`]
+            The object of type that was specified or ``None`` if not found.
+
+        Usage
+        -----
+        ``await GUILD_OBJECT.get_or_fetch(OBJECT_TYPE, OBJECT_ID)``
+        """
+
+        def get_attr_name(t: object_type) -> str:
```

Comment:
What was the point of even including the main portion of the code in a nested function? `get_or_fetch` doesn't seem to take in a function argument that it can use to derive the name of the attribute.